src/ParametricOptInterface.jl

```diff
@@ -636,6 +638,10 @@ function MOI.set(model::ParametricOptimizer, attr::MOI.ObjectiveFunction, f::MOI
     end
 end
 
+function MOI.copy_to(model::ParametricOptimizer, src::MathOptInterface.ModelLike; kws...)
```

Comment:
Added this method to avoid this error but wasn't sure what the sensible thing to do was. When using an NLP the  `BridgeOptimizer` and the `CachingOptimizer` wrap around the `ParametricOptimizer` but it isn't clear if the model inside of the `ParametricOptimizer` is the one that should be wrapped instead. 

```Julia
ERROR: MethodError: no method matching copy_to(::ParametricOptInterface.ParametricOptimizer{Float64, Ipopt.Optimizer}, ::MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.GenericModel{Float64, MathOptInterface.Utilities.ModelFunctionConstraints{Float64}}}; copy_names=false)
Closest candidates are:
  copy_to(::Union{MathOptInterface.Utilities.AbstractModelLike{T}, MathOptInterface.Utilities.AbstractOptimizer{T}} where T, ::MathOptInterface.ModelLike; kws...) at /Users/jdlara/.julia/packages/MathOptInterface/YDdD3/src/Utilities/model.jl:769
  copy_to(::MathOptInterface.Utilities.MockOptimizer, ::MathOptInterface.ModelLike; kws...) at /Users/jdlara/.julia/packages/MathOptInterface/YDdD3/src/Utilities/mockoptimizer.jl:754
  copy_to(::MathOptInterface.Utilities.CachingOptimizer, ::MathOptInterface.ModelLike; kws...) at /Users/jdlara/.julia/packages/MathOptInterface/YDdD3/src/Utilities/cachingoptimizer.jl:223
  ...
Stacktrace:
 [1] attach_optimizer(model::MathOptInterface.Utilities.CachingOptimizer{ParametricOptInterface.ParametricOptimizer{Float64, Ipopt.Optimizer}, MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.GenericModel{Float64, MathOptInterface.Utilities.ModelFunctionConstraints{Float64}}}})
   @ MathOptInterface.Utilities ~/.julia/packages/MathOptInterface/YDdD3/src/Utilities/cachingoptimizer.jl:185
 [2] optimize!(m::MathOptInterface.Utilities.CachingOptimizer{ParametricOptInterface.ParametricOptimizer{Float64, Ipopt.Optimizer}, MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.GenericModel{Float64, MathOptInterface.Utilities.ModelFunctionConstraints{Float64}}}})
   @ MathOptInterface.Utilities ~/.julia/packages/MathOptInterface/YDdD3/src/Utilities/cachingoptimizer.jl:248
 [3] optimize!(b::MathOptInterface.Bridges.LazyBridgeOptimizer{MathOptInterface.Utilities.CachingOptimizer{ParametricOptInterface.ParametricOptimizer{Float64, Ipopt.Optimizer}, MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.GenericModel{Float64, MathOptInterface.Utilities.ModelFunctionConstraints{Float64}}}}})
   @ MathOptInterface.Bridges ~/.julia/packages/MathOptInterface/YDdD3/src/Bridges/bridge_optimizer.jl:319
 [4] optimize!(m::MathOptInterface.Utilities.CachingOptimizer{MathOptInterface.AbstractOptimizer, MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.GenericModel{Float64, MathOptInterface.Utilities.ModelFunctionConstraints{Float64}}}})
   @ MathOptInterface.Utilities ~/.julia/packages/MathOptInterface/YDdD3/src/Utilities/cachingoptimizer.jl:252
 [5] optimize!(model::Model, optimizer_factory::Nothing; bridge_constraints::Bool, ignore_optimize_hook::Bool, kwargs::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
   @ JuMP ~/.julia/packages/JuMP/Xrr7O/src/optimizer_interface.jl:185
 [6] optimize! (repeats 2 times)
   @ ~/.julia/packages/JuMP/Xrr7O/src/optimizer_interface.jl:157 [inlined]
 [7] top-level scope
   @ REPL[79]:1
```

Reply:
Hmm I am not sure why we don`t have a `copy_to` I think there must be something to do with mutating the types of constraints and objectives depending on the location of parameters. @joaquimg

Reply:
Do this instead https://jump.dev/MathOptInterface.jl/dev/tutorials/implementing/#Incremental-and-copy_to.

Your current copy_to won't work.

Reply:
`supports_incremental_interface` isn't defined in MOI 0.9.22

Reply:
@odow if I implement 

```julia
function MOI.copy_to(dest::ParametricOptimizer, src::MOI.ModelLike; kwargs...)
    return MOI.Utilities.automatic_copy_to(dest, src; kwargs...)
end
```

This error appears: 

```julia
ERROR: Model ParametricOptInterface.ParametricOptimizer{Float64, Ipopt.Optimizer} does not support copy.
Stacktrace:
  [1] error(::String, ::String, ::String)
    @ Base ./error.jl:42
  [2] automatic_copy_to(dest::ParametricOptInterface.ParametricOptimizer{Float64, Ipopt.Optimizer}, src::MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.GenericModel{Float64, MathOptInterface.Utilities.ModelFunctionConstraints{Float64}}}; copy_names::Bool, filter_constraints::Nothing)
    @ MathOptInterface.Utilities ~/.julia/packages/MathOptInterface/YDdD3/src/Utilities/copy.jl:28
  [3] copy_to(dest::ParametricOptInterface.ParametricOptimizer{Float64, Ipopt.Optimizer}, src::MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.GenericModel{Float64, MathOptInterface.Utilities.ModelFunctionConstraints{Float64}}}; kwargs::Base.Iterators.Pairs{Symbol, Bool, Tuple{Symbol}, NamedTuple{(:copy_names,), Tuple{Bool}}})
    @ ParametricOptInterface ~/.julia/dev/ParametricOptInterface/src/ParametricOptInterface.jl:653
  [4] attach_optimizer(model::MathOptInterface.Utilities.CachingOptimizer{ParametricOptInterface.ParametricOptimizer{Float64, Ipopt.Optimizer}, MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.GenericModel{Float64, MathOptInterface.Utilities.ModelFunctionConstraints{Float64}}}})
```

Reply:
if I implement: 

```julia
function MOI.copy_to(dest::ParametricOptimizer, src::MOI.ModelLike; kwargs...)
    return MOI.Utilities.automatic_copy_to(dest.optimizer, src; kwargs...)
end
```

then the current error is thrown: 

```julia
ERROR: MathOptInterface.UnsupportedConstraint{MathOptInterface.SingleVariable, ParametricOptInterface.Parameter}: `MathOptInterface.SingleVariable`-in-`ParametricOptInterface.Parameter` constraint is not supported by the model.
Stacktrace:
  [1] correct_throw_add_constraint_error_fallback(model::Ipopt.Optimizer, func::MathOptInterface.SingleVariable, set::ParametricOptInterface.Parameter; error_if_supported::MathOptInterface.AddConstraintNotAllowed{MathOptInterface.SingleVariable, ParametricOptInterface.Parameter})
    @ MathOptInterface ~/.julia/packages/MathOptInterface/YDdD3/src/constraints.jl:199
  [2] correct_throw_add_constraint_error_fallback(model::Ipopt.Optimizer, func::MathOptInterface.SingleVariable, set::ParametricOptInterface.Parameter)
    @ MathOptInterface ~/.julia/packages/MathOptInterface/YDdD3/src/constraints.jl:196
  [3] throw_add_constraint_error_fallback(model::Ipopt.Optimizer, func::MathOptInterface.SingleVariable, set::ParametricOptInterface.Parameter; kwargs::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ MathOptInterface ~/.julia/packages/MathOptInterface/YDdD3/src/constraints.jl:148
  [4] throw_add_constraint_error_fallback(model::Ipopt.Optimizer, func::MathOptInterface.SingleVariable, set::ParametricOptInterface.Parameter)
    @ MathOptInterface ~/.julia/packages/MathOptInterface/YDdD3/src/constraints.jl:148
  [5] add_constraint(model::Ipopt.Optimizer, func::MathOptInterface.SingleVariable, set::ParametricOptInterface.Parameter)
    @ MathOptInterface ~/.julia/packages/MathOptInterface/YDdD3/src/constraints.jl:136
  [6] _broadcast_getindex_evalf
    @ ./broadcast.jl:648 [inlined]
  [7] _broadcast_getindex
    @ ./broadcast.jl:621 [inlined]
  [8] getindex
    @ ./broadcast.jl:575 [inlined]
  [9] copy
    @ ./broadcast.jl:922 [inlined]
 [10] materialize
    @ ./broadcast.jl:883 [inlined]
 [11] add_constraints(model::Ipopt.Optimizer, funcs::Vector{MathOptInterface.SingleVariable}, sets::Vector{ParametricOptInterface.Parameter})
    @ MathOptInterface ~/.julia/packages/MathOptInterface/YDdD3/src/constraints.jl:229
 [12] copy_constraints(dest::Ipopt.Optimizer, src::MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.GenericModel{Float64, MathOptInterface.Utilities.ModelFunctionConstraints{Float64}}}, idxmap::MathOptInterface.Utilities.IndexMap, cis_src::Vector{MathOptInterface.ConstraintIndex{MathOptInterface.SingleVariable, ParametricOptInterface.Parameter}}, filter_constraints::Nothing)
    @ MathOptInterface.Utilities ~/.julia/packages/MathOptInterface/YDdD3/src/Utilities/copy.jl:419
 [13] copy_constraints(dest::Ipopt.Optimizer, src::MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.GenericModel{Float64, MathOptInterface.Utilities.ModelFunctionConstraints{Float64}}}, idxmap::MathOptInterface.Utilities.IndexMap, cis_src::Vector{MathOptInterface.ConstraintIndex{MathOptInterface.SingleVariable, ParametricOptInterface.Parameter}})
    @ MathOptInterface.Utilities ~/.julia/packages/MathOptInterface/YDdD3/src/Utilities/copy.jl:409
 [14] pass_constraints(dest::Ipopt.Optimizer, src::MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.GenericModel{Float64, MathOptInterface.Utilities.ModelFunctionConstraints{Float64}}}, copy_names::Bool, idxmap::MathOptInterface.Utilities.IndexMap, single_variable_types::Vector{DataType}, single_variable_indices::Vector{Vector{T} where T}, vector_of_variables_types::Vector{DataType}, vector_of_variables_indices::Vector{Any}, pass_cons::typeof(MathOptInterface.Utilities.copy_constraints), pass_attr::typeof(MathOptInterface.set); filter_constraints::Nothing)
    @ MathOptInterface.Utilities ~/.julia/packages/MathOptInterface/YDdD3/src/Utilities/copy.jl:502
 [15] default_copy_to(dest::Ipopt.Optimizer, src::MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.GenericModel{Float64, MathOptInterface.Utilities.ModelFunctionConstraints{Float64}}}, copy_names::Bool, filter_constraints::Nothing)
    @ MathOptInterface.Utilities ~/.julia/packages/MathOptInterface/YDdD3/src/Utilities/copy.jl:714
 [16] #automatic_copy_to#127
    @ ~/.julia/packages/MathOptInterface/YDdD3/src/Utilities/copy.jl:24 [inlined]
 [17] #copy_to#3
    @ ~/.julia/dev/ParametricOptInterface/src/ParametricOptInterface.jl:653 [inlined]
 [18] attach_optimizer(model::MathOptInterface.Utilities.CachingOptimizer{ParametricOptInterface.ParametricOptimizer{Float64, Ipopt.Optimizer}, MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.GenericModel{Float64, MathOptInterface.Utilities.ModelFunctionConstraints{Float64}}}})
    @ MathOptInterface.Utilities ~/.julia/packages/MathOptInterface/YDdD3/src/Utilities/cachingoptimizer.jl:185
 [19] optimize!(m::MathOptInterface.Utilities.CachingOptimizer{ParametricOptInterface.ParametricOptimizer{Float64, Ipopt.Optimizer}, MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.GenericModel{Float64, MathOptInterface.Utilities.ModelFunctionConstraints{Float64}}}})
    @ MathOptInterface.Utilities ~/.julia/packages/MathOptInterface/YDdD3/src/Utilities/cachingoptimizer.jl:248
 [20] optimize!(b::MathOptInterface.Bridges.LazyBridgeOptimizer{MathOptInterface.Utilities.CachingOptimizer{ParametricOptInterface.ParametricOptimizer{Float64, Ipopt.Optimizer}, MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.GenericModel{Float64, MathOptInterface.Utilities.ModelFunctionConstraints{Float64}}}}})
    @ MathOptInterface.Bridges ~/.julia/packages/MathOptInterface/YDdD3/src/Bridges/bridge_optimizer.jl:319
 [21] optimize!(m::MathOptInterface.Utilities.CachingOptimizer{MathOptInterface.AbstractOptimizer, MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.GenericModel{Float64, MathOptInterface.Utilities.ModelFunctionConstraints{Float64}}}})
    @ MathOptInterface.Utilities ~/.julia/packages/MathOptInterface/YDdD3/src/Utilities/cachingoptimizer.jl:252
 [22] optimize!(model::Model, optimizer_factory::Nothing; bridge_constraints::Bool, ignore_optimize_hook::Bool, kwargs::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ JuMP ~/.julia/packages/JuMP/Xrr7O/src/optimizer_interface.jl:185
 [23] optimize! (repeats 2 times)
    @ ~/.julia/packages/JuMP/Xrr7O/src/optimizer_interface.jl:157 [inlined]
 [24] top-level scope
    @ REPL[18]:1
```

Reply:
Sorry. Here's the 0.9.22 link https://jump.dev/MathOptInterface.jl/stable/manual/implementing/#Incremental-and-copy_to.

You need to implement `MOI.Utilities.supports_default_copy_to`.

This is the correct one
```julia
function MOI.copy_to(dest::ParametricOptimizer, src::MOI.ModelLike; kwargs...)
    return MOI.Utilities.automatic_copy_to(dest, src; kwargs...)
end
```